jmclient/jmclient/wallet_rpc.py

```diff
@@ -375,21 +375,44 @@ def set_token(self, wallet_name):
         # startup, so any failure to exist here is a logic error:
         self.wss_factory.valid_token = self.cookie
 
-    def get_POST_body(self, request, keys):
+    def get_POST_body(self, request, required_keys, optional_keys=None):
         """ given a request object, retrieve values corresponding
-        to keys keys in a dict, assuming they were encoded using JSON.
-        If *any* of the keys are not present, return False, else
-        returns a dict of those key-value pairs.
+        to keys in a dict, assuming they were encoded using JSON.
+        If *any* of the required_keys are not present, return False, else
+        returns a dict of those key-value pairs and any of the optional key-value pairs.
         """
         assert isinstance(request.content, BytesIO)
         # we swallow any formatting failure here:
         try:
             json_data = json.loads(request.content.read().decode(
                 "utf-8"))
-            return {k: json_data[k] for k in keys}
+            required_body = {k: json_data[k] for k in required_keys}
+
+            if optional_keys is not None:
+                optional_body = {k: json_data[k] for k in optional_keys if k in json_data}
+                # required keys take precedence over optional keys
+                optional_body.update(required_body)
+                return optional_body
```

Comment:
The logic here seems perfectly fine, but I do wonder if it makes sense to error out if the an entry in `optional_keys` actually conflicts with one in `required_keys`, rather than overwrite the former with the latter.

Reply:
Updated. ✅